static/js/src/advantage/manage-subscription.js

```diff
@@ -0,0 +1,256 @@
+          <label for="confirm-cancel">Please type <strong>cancel</strong> to confirm.</label>
+        </div>
+        <div id="cancel-modal-buttons-wrapper" class="u-align--right">
+        </div>
+      </div>
+    </div>
+  `;
+  container.innerHTML = content;
+  document.body.appendChild(container);
+
+  const fieldWrapper = document.getElementById("cancel-modal-field-wrapper");
+  fieldWrapper.appendChild(confirmCancelField);
+
+  const buttonWrapper = document.getElementById("cancel-modal-buttons-wrapper");
+  buttonWrapper.appendChild(goBackButton);
+  buttonWrapper.appendChild(confirmCancelButton);
+}
+
+function handleCancelChangesClick(id) {
+  // Hide edit options
+  const previewSection = document.getElementById(`view-mode--${id}`);
+  const editSection = document.getElementById(`edit-mode--${id}`);
+  previewSection.classList.remove("u-hide");
+  editSection.classList.add("u-hide");
+
+  // Reset input and help text
+  const resizeField = document.getElementById(`resize-input--${id}`);
+  resizeField.oninput = () => {};
+  resizeField.value = resizeField.defaultValue;
+  const resizeSummary = document.getElementById(`resize-summary--${id}`);
+  const newPayment = document.getElementById(`new-payment--${id}`);
+  const updateButton = document.getElementById(`save-changes--${id}`);
+  resizeSummary.classList.add("u-hide");
+  newPayment.classList.add("u-hide");
+  updateButton.disabled = true;
+
+  // Remove buttons handlers
+  const cancelSubscriptionButton = document.getElementById(
+    `cancel-subscription--${id}`
+  );
+  const cancelChangesButton = document.getElementById(`cancel-changes--${id}`);
+  updateButton.onclick = () => {};
+  cancelChangesButton.onclick = () => {};
+  if (cancelSubscriptionButton) cancelSubscriptionButton.onclick = () => {};
+}
+
+function handleChange(e, id) {
+  const defaultValue = Number.parseInt(e.target.defaultValue);
+  let newValue = Number.parseInt(e.target.value);
+
+  const unitPrice = Number.parseFloat(e.target.dataset.unitPrice.split(" ")[0]);
+  const nextPayment = Number.parseFloat(
+    e.target.dataset.nextPayment.split(" ")[0]
+  );
+
+  const billingPeriod = e.target.dataset.billingPeriod;
+
+  const formatter = new Intl.NumberFormat("en-US", {
+    style: "currency",
+    currency: "USD",
+  });
+
+  const resizeSummary = document.getElementById(`resize-summary--${id}`);
+  const newPayment = document.getElementById(`new-payment--${id}`);
+  const updateButton = document.getElementById(`save-changes--${id}`);
+
+  const { min, max } = e.target;
+
+  if (newValue < min) {
+    newValue = Number.parseInt(min);
+    e.target.value = min;
+  }
+
+  if (newValue > max) {
+    newValue = Number.parseInt(max);
+    e.target.value = max;
+  }
+
+  if (newValue !== defaultValue) {
+    resizeSummary.classList.remove("u-hide");
+    newPayment.classList.remove("u-hide");
+    updateButton.disabled = false;
+
+    if (newValue > defaultValue) {
```

Comment:
I think you could do with setting a variable for the increaseAmount as its calculated a few times in this function.

```js
const increaseAmount = newValue - defaultValue;
if (increaseAmount > 0) {
  ...
```

Reply:
I've called it `deltaMachines` because it can be negative. Let me know if the name is confusing